archunit-junit/build.gradle

```diff
@@ -96,4 +96,3 @@ ext.configureJUnitArchive = { configureDependencies ->
 this.with addCleanThirdPartyTask
 
 javadoc.enabled = false
-uploadArchives.enabled = false
```

Comment:
Why has this been removed? Are you sure it's not publishing `archunit-junit` as a separate archive now?

Reply:
When running on Gradle 7.3 I get the following error message is this line is present:
```
> Configure project :archunit-java-modules-test
Project :archunit-java-modules-test => 'example_module' Java module

FAILURE: Build failed with an exception.

* Where:
Build file '/home/tim/Documents/workspace/archunit/archunit-junit/build.gradle' line: 99

* What went wrong:
A problem occurred evaluating project ':archunit-junit'.
> Could not get unknown property 'uploadArchives' for project ':archunit-junit' of type org.gradle.api.Project.
```
Figured as the flag was disabled, not being able to find the property would lead to the same result.
Although some migration might still be in order if it was replaced by another now enabled by default task.

Reply:
Yes, we need to look if it was replaced by a different API then. It would be weird if it was disabled by default, so I would be afraid that it publishes this now as an archive...

Reply:
Added a separate commit with link the the docs:
https://docs.gradle.org/current/userguide/upgrading_version_6.html#removal_of_the_uploadarchives_task
It seems the functionality was moved to plugins, so we're unlikely to inadvertently upload these artifacts now. :)

Reply:
Although I do see the maven-publish plugin used in: `/archunit/build-steps/release/publish.gradle`.
Can't hurt to double check for anyone more familiar with Gradle and artifact publishing.